Enable anonymisation in integration testing tutorial

rnicoll opened the pull request:
Enable anonymisation in integration testing tutorial, and as a requirement fix a bug where the counterparty anonymous identity was not registered by `TransactionKeyFlow`.

Enforces that all parties are identified during `FinalityFlow` to avoid silently failing to deliver transactions to a party.

mnesbit commented on core/src/main/kotlin/net/corda/flows/FinalityFlow.kt:
```diff
@@ -103,8 +104,9 @@ class FinalityFlow(val transactions: Iterable<SignedTransaction>,
-            // TODO: Is it safe to drop participants we don't know how to contact? Does not knowing how to contact them count as a reason to fail?
-            val parties = keys.mapNotNull { serviceHub.identityService.partyFromAnonymous(it) }.toSet()
+            val parties = keys.map { anon ->
+                serviceHub.identityService.partyFromAnonymous(anon) ?: throw IllegalArgumentException("Cannot resolve well known identity of anonymous key ${anon.owningKey.toStringShort()}")
```
Is failing correct here? Surely we must allow transactions with some party keys we don't understand, otherwise there is no anonymisation ever?

rnicoll commented:
Rebased and removed the enforced lookup of parties in `FinalityFlow` as it breaks a bunch of unrelated tests, and I'd like to get this bug fix in ASAP.
